JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/GroupBean.java

```diff
@@ -146,31 +136,160 @@ private int getUserId(String userNickName) {
+            return null;
+        }
+
+        return group;
+    }
+
+    public Group getGroup(String groupName) {
+
+        Group group = null;
+        try {
+            group = (Group) em.createQuery("SELECT g FROM Group g WHERE g.groupName=:groupname")
+                    .setParameter("groupname", groupName)
+                    .getSingleResult();
+        }catch(NoResultException noResult) {
+            logger.log(Level.WARNING, "NO RESULT IN QUERY IN GETGROUP()", noResult);
+            return null;
+        }
+        return group;
+    }
+
+    public int getGroupId(String groupName) {
+
+        Group group = getGroup(groupName);
+        if (group == null) {
+            return -1;
+        }
+        return group.getGroupId();
+
+    }
+
+    public Group createGroup(String groupName) {
+
+        Group g = new Group(groupName);
+
+        em.persist(g);
+        em.flush();
+
+        return g;
+
+
+    }
+
+    public Access createAccess(String accessName) {
+        Access a = new Access(accessName);
+
+        em.persist(a);
+        em.flush();
+
+        return a;
+
+    }
+
+    public UserGroup createUserGroup(User u, Group g, Access a) {
+
+
+/*
+        Access a = (Access)em.createQuery("SELECT a FROM Access a WHERE a.accessId =:accessid")
+                            .setParameter("accessid", accessId)
+                            .getSingleResult();
 
-        Group group = (Group)em.createQuery("SELECT g FROM Group g WHERE g.groupId=:groupid")
+        Group g = (Group)em.createQuery("SELECT g FROM Group g WHERE g.groupId =:groupid")
                 .setParameter("groupid", groupId)
                 .getSingleResult();
-        return group;
+
+        User u = (User)em.createQuery("SELECT u FROM User u WHERE u.userId =:userid")
+                .setParameter("userid", userId)
+                .getSingleResult();*/
+
+        UserGroup ug = new UserGroup(g, u, a);
+
+        ug.setUserId(u.getUserId());
+        ug.setGroupId(g.getGroupId());
+
+        //  em.getTransaction().begin();
+        em.persist(ug);
+        //       em.flush();
+        //  em.getTransaction().commit();
+        return ug;
+
     }
 
-    //!TODO
-    public Map<String, String> getAllMembers(int groupId){
-        return null;
+    public UserGroup getUserGroup(int userId, int groupId) {
+
+        UserGroup ug;
+        try {
+            ug = (UserGroup)em.createQuery("SELECT ug FROM UserGroup ug WHERE ug.userId =:userid AND ug.groupId =:groupid")
+                    .setParameter("userid", userId)
+                    .setParameter("groupid", groupId)
+                    .getSingleResult();
+        } catch (NoResultException noResult) {
+            logger.log(Level.WARNING, "NO RESULT IN QUERY IN getUserGroup()", noResult);
+            return null;
+        }
+
+        return ug;
+    }
+
+    public void addUserGroupToGroupList(Group group, UserGroup userGroup) {
```

Comment:
Another method for associating UserGroup with the Group. I believe there should be only one.